### ISSUE TITLE
Make the `tdigest` public 

### ISSUE THREAD

**Issue by l45k** [apache/arrow-datafusion#8554]
### Is your feature request related to a problem or challenge?

I would like to build a custom aggregator based on the [tdigest](https://github.com/apache/arrow-datafusion/blob/main/datafusion/physical-expr/src/aggregate/tdigest.rs). The tdigest `mod` is private, and thus I can't use it outside DataFusion. It would be great if it would be changed to poublic to allow custom aggregators based on it.

### Describe the solution you'd like

To allow building new aggregators based on the `tdigest` make it public by changing the following
https://github.com/apache/arrow-datafusion/blob/b276d479918400105017db1f7f46dcb67b52206d/datafusion/physical-expr/src/aggregate/mod.rs#L58
to 
```rust
pub mod tdigest
```

### Describe alternatives you've considered

If changing the visibility to public is not possible, I would need to introduce my own tdigest and reproduce a lot of code to make it work as an aggregator. This also means that I need to update the code with each new version of DataFusion.

**Comment by alamb** [comment]
I think making tdigest public would be fine, though likely brittle if we move the implementation of aggregation functions

That being said, I don't think it has changed much since its initial implementation, so you could also potentially just copy/paste it into your codebase ("fork" it).

**Comment by l45k** [comment]
@alamb Thanks for your answer! 

Forking it is my current solution. Although, I am not a big fan of reproducing code since it introduces more complexity and a code path that you need to maintain up to date with the original code. However, if turning the visibility of the TDigest to public degrades datafusion's code base, I will need to keep my fork.

**Comment by alamb** [comment]
I don't think it degrades the codebase, but I also think using it as a public API will subject you to changes as DataFusion evolves. This is likely fine, but I wanted to make sure the tradeoffs were clear

**Comment by rspears74** [comment]
@l45k Not sure if this is helpful or not, but I just forked the MnO3/tdigest library, made a few small changes I needed (like making the struct fields public), and implemented a UDAF.
